### PR TITLE
Account for ConfigItem being generated by another copy of Babel

### PR DESCRIPTION
**Pull request by developit**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | developit/microbundle#657
| Patch: Bug Fix?          | yes
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | 
| Documentation PR Link    | N/A
| Any Dependency Changes?  | no
| License                  | MIT

This fixes `ConfigItem` unboxing when `core.createConfigItem()` is used to generate a ConfigItem that is then passed to a different copy of Babel. This is common in cases where npm fails to dedupe modules, or when differing versions of Babel are in-use.

```
// good - uses the version of babel that is doing the transforming:
module.exports = function preset(api) {
  api.createConfigItem(...)
}
// bad - uses whatever version this module installed:
import api from '@babel/core';
module.exports = babelLoader.custom({
  options() {
    api.createConfigItem(..)
  }
});
```

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/11734"></a>

### CODE REVIEW

**Comment by developit** [comment]
Another suggestion, rather than changing all the instanceof checks, might be to throw or warn when this behavior is detected:

```js
import core from '@babel/core';  // could be a separate copy of Babel (determined by npm)

export function custom() {
  return {
    config() {
      core.createConfigItem(...)
      // "Error: detected ConfigItem instances from a different version of Babel"
    }
  }
}
```

File: packages/babel-core/src/config/item.js [hunk 1]
```diff
@@ -64,6 +64,11 @@ class ConfigItem {
    */
   _descriptor: UnloadedDescriptor;
 
+  /**
+   * Used to detect ConfigItem instances from other Babel instances.
+   */
+  _isConfigItem: true;
```
Comment (nicolo-ribaudo) [inline packages/babel-core/src/config/item.js:70]:
```suggestion
  [Symbol.for("@babel/core@7 - ConfigItem")]: true;
```

And we can remove the `instanceof` check and only rely on this.

File: packages/babel-core/src/config/item.js [hunk 2]
```diff
@@ -40,7 +40,7 @@ export function createConfigItem(
 }
 
 export function getItemDescriptor(item: mixed): UnloadedDescriptor | void {
-  if (item instanceof ConfigItem) {
+  if (item && item[Symbol.for("@babel/core@7 - ConfigItem")] === true) {
```
Comment (JLHwung) [inline packages/babel-core/src/config/item.js:43]:
```suggestion
  if (item?.[Symbol.for("@babel/core@7 - ConfigItem")] === true) {
```

File: packages/babel-core/src/config/item.js [hunk 3]
```diff
@@ -64,6 +64,11 @@ class ConfigItem {
    */
   _descriptor: UnloadedDescriptor;
 
+  /**
```
Comment (nicolo-ribaudo) [inline packages/babel-core/src/config/item.js:67]:
```suggestion
  // TODO(Babel 8): Check if this symbol needs to be updated
  /**
```

**Review by nicolo-ribaudo — APPROVED** [review]

**Review by existentialism — APPROVED** [review]

**Pull request merged**